### PR TITLE
Added Apache thrift support

### PR DESCRIPTION
**Pull request by JonnoFTW**
per #156 I have separated them into two branches. This branch implements apache's json protocol in  thriftpy2

### CODE REVIEW

**Comment by JonnoFTW** [comment]
@ethe can you please take a look at this?

**Comment by ethe** [comment]
thank you, i will review it as soon as possible.

File: thriftpy2/protocol/apache_json.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        Only really used to read the arguments off a request into whatever we want
+        see thriftpy2.thrift.TProcessor.process_in for how this class will be used
+
+        Will turn the contents of self.req[4] into the args of obj,
+        ie. self.req[4]["1"] must be rendered into obj.thrift_spec
+
+        :param obj:
+        :return:
+        """
+        return self._dict_to_thrift(self._req[4], obj)
```
Comment (ethe) [inline thriftpy2/protocol/apache_json.py:309]: it is better to add a new line at the end of the file?

**Comment by ethe** [comment]
LGTM, please fix the above comment

File: thriftpy2/protocol/apache_json.py (hunk 1 above)
Comment (JonnoFTW) [reply thriftpy2/protocol/apache_json.py:309]: Done

**Comment by ethe** [comment]
There are some conflicts between #164 and it

**Review by ethe — APPROVED** [review]

**Pull request merged**